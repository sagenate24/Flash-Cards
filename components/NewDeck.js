import React, { Component } from 'react';
import {
  Text,
  View,
  TextInput,
  StyleSheet,
} from 'react-native';
import { connect } from 'react-redux';
import { addDeckTitle, getDeck } from '../utils/api';
import { white, black, red } from '../utils/colors';
import { addDeck } from '../actions/decks';

import NASBtn from './NASBtn';

class NewDeck extends Component {
  state = {
    title: '',
    underColorT: false,
  };

  submit = () => {
    const { title } = this.state;
    const { dispatch, navigation } = this.props;

    addDeckTitle(title).then(() => getDeck(title).then((deck) => {
      dispatch(addDeck({
        [deck.title]: deck,
      }));
      navigation.navigate(
        'Deck',
        { currentDeck: deck },
      );
    }));
  };

  render() {
    const { title, underColorT } = this.state;
    const charactersLeft = 27 - title.length;

    return (
      <View style={styles.container}>
        <View style={styles.formPage}>
          <TextInput
            value={title}
            selectionColor={black}
            underlineColorAndroid="rgba(0,0,0,0)"
            maxLength={27}
            onChangeText={title => this.setState({ title })}
            onFocus={() => this.setState({ underColorT: true })}
            style={underColorT === true ? styles.inputActive : styles.input}
          />
          <Text>TITLE</Text>
          {charactersLeft <= 22 && (
            <Text style={styles.inputIsGettingFull}>{charactersLeft}</Text>
          )}
        </View>
        <NASBtn tintColor={{ backgroundColor: red }} disabled={title === ''} onPress={this.submit}>
          Create Deck
        </NASBtn>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 20,
  },
  formPage: {
    backgroundColor: white,
    borderRadius: 2,
    padding: 20,
    shadowOpacity: 0.8,
    marginBottom: 10,
    shadowColor: 'rgba(0,0,0,0.24)',
    shadowOffset: {
      width: 0,
      height: 3,
    },
  },
  input: {
    fontSize: 20,
    paddingBottom: 2,
    borderBottomWidth: 2,
    borderColor: black,
  },
  inputActive: {
    fontSize: 20,
    borderBottomWidth: 4,
    borderColor: '#e6b800',
  },
  inputIsGettingFull: {
    color: red,
    fontSize: 16,
    opacity: 0.9,
  },
});

export default connect()(NewDeck);
