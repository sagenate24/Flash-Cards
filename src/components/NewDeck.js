import React, { Component } from 'react';
import {
  Text,
  View,
  TextInput,
  StyleSheet,
  Alert,
} from 'react-native';
import { connect } from 'react-redux';
import { addDeck } from '../actions/decks';
import { addDeckTitle, getDeck } from '../utils/api';
import { white, black, red } from '../utils/colors';

import NASBtn from './NASBtn';

class NewDeck extends Component {
  state = {
    title: '',
    underColorT: false,
  };

  submit = () => {
    const { title } = this.state;
    const { dispatch, navigation, deckTitleList } = this.props;
    const matchedTitle = deckTitleList.find(deckTitle => deckTitle === title);

    if (matchedTitle !== title) {
      addDeckTitle(title).then(() => getDeck(title).then((deck) => {
        dispatch(addDeck({
          [deck.title]: deck,
        }));
        navigation.navigate(
          'Deck',
          { currentDeck: deck },
        );
      }));

      this.setState(() => ({
        title: '',
      }));
    } else {
      Alert.alert(
        'Warning',
        'You already have a deck with this title. Please choose another.',
        [
          { text: 'OK' },
        ],
        { cancelable: false },
      );
    }
  };

  render() {
    const { title, underColorT } = this.state;
    const charactersLeft = 27 - title.length;

    return (
      <View style={styles.container}>
        <View style={styles.formPage}>
          <TextInput
            value={title}
            maxLength={27}
            selectionColor={black}
            underlineColorAndroid="rgba(0,0,0,0)"
            onChangeText={title => this.setState({ title })}
            onFocus={() => this.setState({ underColorT: true })}
            style={underColorT === true ? styles.inputActive : styles.input}
          />
          <Text>TITLE</Text>
          {charactersLeft <= 22 && (
            <Text style={styles.inputIsGettingFull}>{charactersLeft}</Text>
          )}
        </View>
        <NASBtn tintColor={{ backgroundColor: red }} disabled={title === ''} onPress={this.submit}>
          Create Deck
        </NASBtn>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 20,
  },
  formPage: {
    backgroundColor: white,
    borderRadius: 2,
    padding: 20,
    shadowOpacity: 0.8,
    marginBottom: 10,
    shadowColor: 'rgba(0,0,0,0.24)',
    shadowOffset: {
      width: 0,
      height: 3,
    },
  },
  input: {
    fontSize: 20,
    paddingBottom: 2,
    borderBottomWidth: 2,
    borderColor: black,
  },
  inputActive: {
    fontSize: 20,
    borderBottomWidth: 4,
    borderColor: '#e6b800',
  },
  inputIsGettingFull: {
    color: red,
    fontSize: 16,
    opacity: 0.9,
  },
});

function mapStateToProps({ decks, profile }) {
  return {
    parentalControl: profile.parentControl,
    deckTitleList: Object.keys(decks),
  };
}

export default connect(mapStateToProps)(NewDeck);
