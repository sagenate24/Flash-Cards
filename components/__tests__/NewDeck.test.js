import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import NewDeck from '../NewDeck';

test('NewDeck snapShot', () => {
  const snap = renderer.create(
    <NewDeck.WrappedComponent />,
  ).toJSON();

  expect(snap).toMatchSnapshot();
});
