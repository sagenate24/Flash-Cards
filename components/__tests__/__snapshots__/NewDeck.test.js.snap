// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewDeck snapShot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "marginBottom": 10,
        "padding": 20,
        "shadowColor": "rgba(0,0,0,0.24)",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.8,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      maxLength={27}
      onChangeText={[Function]}
      onFocus={[Function]}
      selectionColor="#404040"
      style={
        Object {
          "borderBottomWidth": 2,
          "borderColor": "#404040",
          "fontSize": 20,
          "paddingBottom": 2,
        }
      }
      underlineColorAndroid="rgba(0,0,0,0)"
      value=""
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      TITLE
    </Text>
  </View>
  <View
    style={
      Object {
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 10,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#e63946",
          "borderRadius": 0,
          "opacity": 1,
          "padding": 10,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
            "fontSize": 22,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Create Deck
      </Text>
    </View>
  </View>
</View>
`;
